daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md

```diff
@@ -0,0 +1,155 @@
+---
+type: docs
+title: "HashiCorp Consul"
+linkTitle: "HashiCorp Consul"
+description: Detailed information on the HashiCorp Consul name resolution component
+--- 
+
+## Configuration format
+
+Hashicorp Consul is setup within the [Dapr Configuration]({{< ref configuration-overview.md >}}).
+
+Within the config, add a `nameResolution` spec and set the `component` field to `"consul"`.
+
+If you are using the Dapr sidecar to register your service to Consul then you will need the following configuration:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: appconfig
+spec:
+  nameResolution:
+    component: "consul"
+    configuration:
+      selfRegister: true
+```
+
+If Consul service registration is managed externally from Dapr you need to ensure that the Dapr-to-Dapr internal gRPC port is added to the service metadata under `DAPR_PORT` (this key is configurable) and that the Consul service Id matches the Dapr app Id. You can then omit `selfRegister` from the config above.
+
+## Behaviour
+
+On init the Consul component will either validate the connection to the configured (or default) agent or register the service if configured to do so. The name resolution interface does not cater for an "on shutdown" pattern so please consider this if using Dapr to register services to Consul as it will not deregister services.
+
+The component resolves target apps by filtering healthy services and looks for a `DAPR_PORT` in the metadata (key is configurable) in order to retrieve the Dapr sidecar port. Consul `service.meta` is used over `service.port` so as to not interfere with existing Consul estates.
+
+
+## Spec configuration fields
+
+As of writing the configuration spec is fixed to v1.3.0 of the Consul api
+
+| Field          | Required | Type              | Details      |
+| :------------ |:--------:|------------------:| :----------------|
```

Comment:
Please add a column for example, & include an example of what a user may enter

Reply:
I figured the examples on the bottom of the page would do, otherwise do you have any suggestions on how I can include a code block in a table cell?

Reply:
Examples at the bottom of the page were fantastic! For this I was thinking more of single field examples to show how to specify the value. Like https://github.com/dapr/docs/blob/v1.1/daprdocs/content/en/reference/components-reference/supported-secret-stores/hashicorp-vault.md#spec-metadata-fields

Helps users to know if they need quotes or brackets

Reply:
I understand, thing is though some of the fields are complex objects (unlike other components) so I can't simply add a single value for the example column, I need to somehow elegantly display the example object...

Reply:
Good call, for all multi-lines let's reference the sample configs. Adding a suggestion. Also, does SelfRegister default to true?